src/main/java/net/fabricmc/fabric/impl/launch/APIWarning.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.launch;
+
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import javax.swing.JOptionPane;
+
+public final class APIWarning {
+	public static void main(String[] args) {
+		Locale defaultLocale = Locale.getDefault();
+		ResourceBundle WarningAPI = ResourceBundle.getBundle("lang/WarningAPI", defaultLocale);
+		JOptionPane.showMessageDialog(null, WarningAPI.getString("api.warning"));
```

Comment:
This will fail if fabric api is run in a headless environment. Just check `GraphicsEnvironment` to validate that otherwise print the message in sys out.

Reply:
Would system error be better? Since it's already an error, and this way it will be highlighted red (in consoles/terminals that treat error differently to out).

Reply:
this has been fixed with Alex's PR